src/anomalib/post_processing/visualizer.py

```diff
@@ -26,29 +25,43 @@
 )
 
 
-@dataclass
 class ImageResult:
     """Collection of data needed to visualize the predictions for an image."""
 
-    image: np.ndarray
-    pred_score: float
-    pred_label: str
-    anomaly_map: np.ndarray | None = None
-    gt_mask: np.ndarray | None = None
-    pred_mask: np.ndarray | None = None
-    gt_boxes: np.ndarray | None = None
-    pred_boxes: np.ndarray | None = None
-    box_labels: np.ndarray | None = None
-
-    heat_map: np.ndarray = field(init=False)
-    segmentations: np.ndarray = field(init=False)
-    normal_boxes: np.ndarray = field(init=False)
-    anomalous_boxes: np.ndarray = field(init=False)
-
-    def __post_init__(self) -> None:
+    def __init__(
+        self,
+        image: np.ndarray,
+        pred_score: float,
+        pred_label: str,
+        anomaly_map: np.ndarray | None = None,
+        gt_mask: np.ndarray | None = None,
+        pred_mask: np.ndarray | None = None,
+        gt_boxes: np.ndarray | None = None,
+        pred_boxes: np.ndarray | None = None,
+        box_labels: np.ndarray | None = None,
+        normalize: bool = False,
+    ):
+        self.image = image
+        self.pred_score = pred_score
+        self.pred_label = pred_label
+        self.anomaly_map = anomaly_map
+        self.gt_mask = gt_mask
+        self.pred_mask = pred_mask
+        self.gt_boxes = gt_boxes
+        self.pred_boxes = pred_boxes
+        self.box_labels = box_labels
+        self.normalize = normalize
+
+        self.heat_map: np.ndarray
+        self.segmentations: np.ndarray
+        self.normal_boxes: np.ndarray
+        self.anomalous_boxes: np.ndarray
+        self.initialize()
+
+    def initialize(self) -> None:
```

Comment:
Do we need `initialize` method? Maybe we could just generate this stuff in the `__init__`. Alternatively, we could rename `initialize` method to a more descriptive name.